Formula/pypy.rb

```diff
@@ -55,7 +58,19 @@ class Pypy < Formula
     sha256 "6773934e5f5fc3eaa8c5a44949b5b924fc122daa0a8aa9f80c835b4ca2a543fc"
   end
 
+  # Build fixes:
+  # - Disable Linux tcl-tk detection since the build script only searches system paths.
+  #   When tcl-tk is not found, it uses unversioned `-ltcl -ltk`, which breaks build.
+  # - Disable building cffi imports with `--embed-dependencies`, which compiles and
+  #   statically links a specific OpenSSL version.
+  # - Add flag `--no-make-portable` to package.py so that we can disable portable build.
+  #   Portable build is default on macOS and copies tcl-tk/sqlite dylibs into bottle.
+  patch :DATA
```

Comment:
Let's upstream this. Or part of it, at the very least.

Reply:
From patch only the 3rd is probably possible to upstream as is:
1. `tcl-tk` detection will need a logic rewrite. Currently can try opening an issue if there isn't one. MacPorts has also been using a patch https://github.com/macports/macports-ports/blob/master/lang/pypy/files/paths.diff
2. The forced embedded dependencies would also need a logic rewrite to expose an argument so user can control setting. They do have a "packaging"-only environment variable `PYPY_NO_EMBED_DEPENDENCIES `, so having a fix to make use of this during build would be one approach (https://doc.pypy.org/en/latest/whatsnew-pypy2-7.3.0.html).
3. The missing `--no-make-portable` can be upstreamed as the logic was designed to expose that to users.

Reply:
Noting that I created an upstream issue with all items: https://foss.heptapod.net/pypy/pypy/-/issues/3538

Probably will impact `pypy3`. If I decide to work on fixing Linux build there, I'll add link to upstream issue.